### ISSUE TITLE
test_type_cast_decimal_from_invalid_string fails with Ruby 2.6.0 rc2

### ISSUE THREAD

**Issue by yahonda** [rails/rails#34719]
### Steps to reproduce
* Install Ruby 2.6.0 rc2
```ruby
$ cd activemodel
$ bundle exec ruby -w -Itest test/cases/type/decimal_test.rb -n test_type_cast_decimal_from_invalid_string
```

### Expected behavior
It should pass.

### Actual behavior
```ruby
$ cd activemodel
$ bundle exec ruby -w -Itest test/cases/type/decimal_test.rb -n test_type_cast_decimal_from_invalid_string
Run options: -n test_type_cast_decimal_from_invalid_string --seed 9399

# Running:

F

Failure:
ActiveModel::Type::DecimalTest#test_type_cast_decimal_from_invalid_string [test/cases/type/decimal_test.rb:18]:
Expected: 0.1e1
  Actual: 0.0


rails test test/cases/type/decimal_test.rb:15



Finished in 0.001427s, 700.9408 runs/s, 1401.8816 assertions/s.
1 runs, 2 assertions, 1 failures, 0 errors, 0 skips
$
```

### System configuration
**Rails version**:master branch

**Ruby version**:ruby 2.6.0rc2 (2018-12-15 trunk 66408) [x86_64-linux]
```ruby
$ gem list |grep bigdecimal
bigdecimal (default: 1.4.0.pre.20181214a)
```

**Comment by y-yagi** [comment]
Maybe related to https://bugs.ruby-lang.org/issues/15426 
If that bug fixes, `"1ignore".to_d` return the same value as BigDecimal 1.3 and `test_type_cast_decimal_from_invalid_string` will pass.

**Comment by yahonda** [comment]
Thanks for the information. Will see when https://bugs.ruby-lang.org/issues/15426 fixed.

**Comment by y-yagi** [comment]
I confirmed this fiexes with r66473. 

```
$ bundle exec ruby -v -w -Itest test/cases/type/decimal_test.rb -n test_type_cast_decimal_from_invalid_string
ruby 2.6.0dev (2018-12-21 trunk 66474) [x86_64-linux]
Run options: -n test_type_cast_decimal_from_invalid_string --seed 15751

# Running:

.

Finished in 0.001719s, 581.8717 runs/s, 2327.4867 assertions/s.
1 runs, 4 assertions, 0 failures, 0 errors, 0 skips 
```

**Comment by yahonda** [comment]
Confirmed this issue is addressed using `ruby 2.6.0dev (2018-12-21 trunk 66474) [x86_64-linux]`

**Comment by yahonda** [comment]
Thanks. I'll keep in mind to report to the bugs.ruby-lang.org in case `ruby-dev` changes some behavior at Rails.
